src/pybind/mgr/zabbix/module.py

```diff
@@ -0,0 +1,209 @@
+                                                             hostname)
+
+            proc = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                                    stdout=subprocess.PIPE)
+
+            for key, value in data.items():
+                proc.stdin.write('{0} ceph.{1} {2}\n'.format(hostname, key,
+                                                             str(value)))
+
+            stdout, stderr = proc.communicate()
+            if proc.returncode != 0:
+                raise RuntimeError('%s exited non-zero' % self.sender)
+
+            self.log.debug('Zabbix Sender: %s', stdout.rstrip())
+        except:
+            raise
+
+
+class Module(MgrModule):
+    run = False
+
+    ceph_health_mapping = {'HEALTH_OK': 0, 'HEALTH_WARN': 1, 'HEALTH_ERR': 2}
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self.q = Queue()
+
+    def get_localized_config(self, key):
+        r = self.get_config(self.get_mgr_id() + '/' + key)
+        if r is None:
+            r = self.get_config(key)
+        return r
+
+    def get_module_config(self):
+        config = dict()
+
+        config['zabbix_sender'] = self.get_localized_config('zabbix_sender') or '/usr/bin/zabbix_sender'
+
+        zabbix_host = self.get_localized_config('zabbix_host')
+        if not zabbix_host:
+            raise RuntimeError(
+                'no zabbix_host configured; try "ceph config-key put mgr/zabbix/zabbix_host <hostname>"')
+
+        config['zabbix_host'] = zabbix_host
+
+        try:
+            zabbix_port = self.get_localized_config('zabbix_port') or '10051'
+            config['zabbix_port'] = int(zabbix_port)
+        except (ValueError, TypeError):
+            raise RuntimeError(
+                'invalid zabbix_port configured. Please specify a valid integer')
+
+        hostname = self.get_localized_config('hostname')
+        if not hostname:
+            raise RuntimeError(
+                'no hostname configured; try "ceph config-key put mgr/zabbix/hostname <hostname>"')
+
+        config['hostname'] = hostname
+
+        try:
+            interval = self.get_localized_config('interval') or '60'
+            config['interval'] = int(interval)
+        except (ValueError, TypeError):
+            raise RuntimeError(
+                'invalid interval configured. Please specify a valid integer')
+
+        if config['interval'] < 10:
+            raise RuntimeError('interval should be set to at least 10 seconds')
+
+        return config
+
+    def get_data(self):
+        data = dict()
+
+        try:
+            health = json.loads(self.get('health')['json'])
+            data['overall_status'] = health['overall_status']
+            for key, value in self.ceph_health_mapping.items():
+                if data['overall_status'] == key:
+                    data['overall_status_int'] = value
+                    break
+        except (KeyError, ValueError):
```

Comment:
Because this module is in-tree, it shouldn't need any of these KeyError/ValueError exception handlers -- if the format of the data structures changes, the module should be updated at the same time.